src/pure_type.erl

```diff
@@ -19,11 +19,40 @@
 %%
 %% -------------------------------------------------------------------
 
+%% @doc Inspired by both the Lasp data type behaviour and the Riak data
+%%      type behaviour.
+
 -module(pure_type).
--author("Vitor Enes Duarte <vitorenesduarte@gmail.com>").
+-author("Georges Younes <georges.r.younes@gmail.com>").
+
+%-export([update/3]).
+-export_type([polog/0, version_vector/0, element/0]).
 
--export_type([pure_type/0]).
 
 %% Define some initial types.
--type pure_type() ::  pure_gcounter |
-                      pure_gset.
+-type pure_type() :: pure_gcounter | pure_pncounter | pure_gset | pure_twopset | pure_aworset.
+-type polog() :: orddict:orddict().
+-type pure_payload() :: {polog(), term()}.
+-type version_vector() :: orddict:orddict().
+-type pure_crdt() :: {pure_type(), pure_payload()}.
+-type pure_operation() :: term().
+-type element() :: term().
+-type value() :: term().
+-type error() :: term().
+
+%% Initialize a Pure op-based CRDT.
+-callback new() -> pure_crdt().
+
+%% Unified interface for allowing parameterized Pure op-based CRDTs.
+-callback new([term()]) -> pure_crdt().
+
+%% Perform an update.
+-callback update(pure_operation(), version_vector(), pure_crdt()) ->
```

Comment:
Every data type should have behaviour type. There's no specific function for pure op based DTs right? So, pure_type won't be a behaviour for now.